### PR TITLE
Update the quickstart for new APIs and improve the style

### PR DESCRIPTION
**Pull request by gvdongen**
- Introduced stepper component to make it easier to parse visually
- Added Kotlin quickstart
- Changed for the 0.9 release changes
- Updated CLI download instructions for the binaries download

TO DO: I think we will need to do another pass over the TS one, once the template has been updated.

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
In the page "welcome to restate" (overview) you have logos per language, can we add kotlin too?

File: docs/get_started/quickstart.mdx [hunk 1]
```diff
@@ -7,25 +7,23 @@ description: ""
...
+import Step from "../../src/components/Stepper";
 
 # Quickstart
 This guide takes you through your first steps with Restate.
 
-## Prerequisites
+<Admonition type="note" title="Prerequisites">
 - Build tools for your programming language:
     - TypeScript: Latest stable version of [NodeJS](https://nodejs.org/en/) >= v18.17.1 and [npm CLI](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) >= 9.6.7
     - Java: [JDK](https://whichjdk.com/) >= 17
```
Comment (slinkydeveloper) [inline docs/get_started/quickstart.mdx:18]: Minimum is `11`. Also maybe write `Java/Kotlin`

File: docs/get_started/quickstart.mdx [hunk 2]
```diff
@@ -39,122 +37,299 @@ Get all the dependencies and build tools:
...
+
+<Step stepLabel="5" title="Send requests to the service">
+
+Invoke the function via:
 
 ```shell
-npm run build
-npm run app
+curl localhost:8080/myservice/hello -H 'content-type: application/json' \
+    -d '{"name": "Restate"}'
```
Comment (slinkydeveloper) [inline docs/get_started/quickstart.mdx:118]: This looks incorrect? can it be the input just the json string, like for java/kotlin? Also isn't the template creating a virtual object?

Perhaps we need to update this template.

**Review by slinkydeveloper — APPROVED** [review]
This looks terrific!

File: docs/get_started/quickstart.mdx (hunk 2 above)
Comment (gvdongen) [reply docs/get_started/quickstart.mdx:118]: Indeed, I mentioned in the PR description that we need to give the TS quickstart another pass after we update the TS template ;)

**Comment by gvdongen** [comment]
> In the page "welcome to restate" (overview) you have logos per language, can we add kotlin too?

Will do it in another PR. There are a few other finetunings I want to do on the overview page

**Pull request merged**